subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -58,6 +58,31 @@ Usage of invalid Java toolchain specifications is no longer supported.
 Related build errors can be avoided by making sure that language version is set on all toolchain specifications.
 See <<toolchains#sec:configuring_toolchain_specifications,user manual>> for more information.
 
+==== Changing test framework after setting test framework options is now an error
+
+When configuring the built-in test task for Java/Groovy/Scala projects, Gradle no longer allows you to
+change the test framework used by the `Test` task after configuring options. This was deprecated and silently
+discarded configuration in some cases.
+
+This now produces an error:
+```
+test {
+   options {
+   }
+
+   useJUnitPlatform()
+}
+```
+You should either set the test framework before configuring options or look into migrating to the <<jvm_test_suite_plugin#jvm_test_suite_plugin,JVM Test Suite Plugin>>.
```

Comment:
```suggestion
Instead, you can:

- set the test framework before configuring options
- migrate to the <<jvm_test_suite_plugin#jvm_test_suite_plugin,JVM Test Suite Plugin>>
```

Suggestion: convert this list of alternatives to a bulleted list for visibility and scannability. Introduce that list with "instead" to make it more obvious that this is an alternate example to the previous counterexample.